src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqBurdenEstimateRepository.kt

```diff
@@ -215,6 +217,59 @@ class JooqBurdenEstimateRepository(
         return mapper.mapBurdenEstimateSets(records)
     }
 
+    override fun getBurdenEstimateOutcomesSequence(groupId: String, touchstoneVersionId: String, scenarioId: String, burdenEstimateSetId: Int)
+            : Sequence<BurdenEstimateOutcome>
+    {
+        //check that the burden estimate set exists in the group etc
+        val set = getBurdenEstimateSet(groupId, touchstoneVersionId, scenarioId, burdenEstimateSetId)
+
+        if (set.isStochastic())
+        {
+            throw InvalidOperationError("Cannot get burden estimate data for stochastic burden estimate sets")
+        }
+
+        return  dsl.select(
+                    DISEASE.ID,
+                    BURDEN_ESTIMATE.YEAR,
+                    BURDEN_ESTIMATE.AGE,
+                    COUNTRY.ID,
+                    COUNTRY.NAME,
+                    BURDEN_OUTCOME.CODE,
+                    BURDEN_ESTIMATE.VALUE
+                )
+                .fromJoinPath(BURDEN_ESTIMATE_SET, BURDEN_ESTIMATE)
+                .join(RESPONSIBILITY)
+                .on(BURDEN_ESTIMATE_SET.RESPONSIBILITY.eq(RESPONSIBILITY.ID))
+                .joinPath(RESPONSIBILITY, SCENARIO, SCENARIO_DESCRIPTION, DISEASE)
+                .join(COUNTRY)
+                .on(BURDEN_ESTIMATE.COUNTRY.eq(COUNTRY.NID))
+                .joinPath(BURDEN_ESTIMATE, BURDEN_OUTCOME)
+                .where(BURDEN_ESTIMATE_SET.ID.eq(burdenEstimateSetId))
+                .orderBy(
+                        DISEASE.ID,
+                        BURDEN_ESTIMATE.YEAR,
+                        BURDEN_ESTIMATE.AGE,
+                        COUNTRY.ID,
+                        COUNTRY.NAME,
+                        BURDEN_OUTCOME.CODE)
+                .fetchInto(BurdenEstimateOutcome::class.java)
+                .asSequence()
```

Comment:
I don't think there's any point calling `asSequence` here, as in the next step the sequence is evaluated anyway in order to group. But I think there's some advantage with a change to the order of operations. `fetchInto` is equivalent to one `fetch()` which returns a list followed by one `map{it.into(...)}` which also returns a list. If the order was `fetch().asSequence().map{it.into(...)}` then the last `map` operation would return a sequence, meaning one less in memory copy. Or even better `fetchSequence().map...`, then no intermediate data structures I think. 

It would be nice to have some way of benchmarking this (and other operations with large datasets, e.g.coverage data) - maybe something to think about.

Also all this reminds me that we're super close to completing this ticket, not sure why we haven't done it yet, we should bring it in to the next sprint! https://vimc.myjetbrains.com/youtrack/issue/VIMC-897